java/code/src/com/redhat/rhn/manager/system/entitling/SystemUnentitler.java

```diff
@@ -72,28 +63,6 @@ public void removeServerEntitlement(Server server, Entitlement ent) {
         else {
             unentitleServer(server, ent);
         }
-
-        server.asMinionServer().ifPresent(s -> {
-            ServerGroupManager.getInstance().updatePillarAfterGroupUpdateForServers(Arrays.asList(s));
```

Comment:
this line seems to be removed but not added anywhere else again. Was this on purpose?

Reply:
Good catch! I removed it on purpose as I wanted to understand if it is really needed and also in which situation exactly. Will do more testing and investigation and re-add the pillar update if I find that it breaks something.

Reply:
With my testing so far I could not find a regression from removing this line (pillar data regeneration). After @chiaradiamarcelo's latest refactoring the method name (`updatePillarAfterGroupUpdateForServers()`) also does no longer match this situation. I will do one more test before merging this patch, maybe this affects monitoring enablement during registration.